src/extract_gpuinfo_intel_i915.c

```diff
@@ -0,0 +1,295 @@
+/*
+ *
+ * Copyright (C) 2022 Maxime Schmitt <maxime.schmitt91@gmail.com>
+ *
+ * This file is part of Nvtop and adapted from igt-gpu-tools from Intel Corporation.
+ *
+ * Nvtop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Nvtop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with nvtop.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "nvtop/device_discovery.h"
+#include "nvtop/extract_gpuinfo_common.h"
+#include "nvtop/extract_processinfo_fdinfo.h"
+#include "nvtop/time.h"
+
+#include "extract_gpuinfo_intel.h"
+
+#include <assert.h>
+#include <drm/drm.h>
+#include <drm/i915_drm.h>
```

Comment:
This would require the linux userspace headers to be installed (`linux-libc-dev`), is that okay?

Reply:
It's the same requirement as for AMD's `#include <libdrm/amdgpu.h>` we need for AMDGPU so nothing new as far as I know :wink: